### PR TITLE
Move to Ktor Cookie handle, Part 2

### PR DESCRIPTION
**Pull request by VDostoyevskiy**

### CODE REVIEW

**Comment by VDostoyevskiy** [comment]
@FooIbar 
Consider rewrite CloudflareInterceptor to KtorPlugin?

**Comment by FooIbar** [comment]
> @FooIbar Consider rewrite CloudflareInterceptor to KtorPlugin?

Honestly, might just drop it.
Haven't got a single report about it's working / not working since added.

**Comment by VDostoyevskiy** [comment]
Will continue to cleanup okhttp usage(url builder)
And introduce high performance Ktor cronet engine in next CL

**Comment by VDostoyevskiy** [comment]
@FooIbar Please check the cookie handle code in case I misunderstand anything

File: app/src/main/kotlin/com/hippo/ehviewer/client/EhCookieStore.kt [hunk 1]
```diff
@@ -55,16 +28,13 @@ object EhCookieStore : CookieJar {
...
-        name(KEY_CONTENT_WARNING)
-        value(CONTENT_WARNING_NOT_SHOW)
-        domain(EhUrl.DOMAIN_E)
-        path("/")
-        expiresAt(Long.MAX_VALUE)
-    }.build()
+    private val sTipsCookie = Cookie(
+        name = KEY_CONTENT_WARNING,
+        value = CONTENT_WARNING_NOT_SHOW,
+    )
```
Comment (FooIbar) [inline app/src/main/kotlin/com/hippo/ehviewer/client/EhCookieStore.kt:34]: Should set expires?

Comment (VDostoyevskiy) [reply app/src/main/kotlin/com/hippo/ehviewer/client/EhCookieStore.kt:34]: > Should set expires?

We don't save it in Webkit CookieManager, but add it dynamically when load cookies for url
I think no need

Comment (FooIbar) [reply app/src/main/kotlin/com/hippo/ehviewer/client/EhCookieStore.kt:34]: Here though https://github.com/FooIbar/EhViewer/blob/5bd0f5f97b3a16cad0c4242fb9922c75618066be/app/src/main/kotlin/com/hippo/ehviewer/client/EhCookieStore.kt#L49-L53

Comment (VDostoyevskiy) [reply app/src/main/kotlin/com/hippo/ehviewer/client/EhCookieStore.kt:35]: Done

**Review by FooIbar — APPROVED** [review]

File: app/src/main/kotlin/com/hippo/ehviewer/client/EhCookieStore.kt [hunk 2]
```diff
@@ -76,33 +47,41 @@ object EhCookieStore : CookieJar {
...
+    fun load(url: Url): List<Cookie> {
+        val cookies = manager.getCookie(url.toString()) ?: return emptyList()
         val checkTips = EhUrl.DOMAIN_E in url.host
-        return get(url).run {
-            if (checkTips) {
-                fastFilter { it.name != KEY_CONTENT_WARNING }.toMutableList().apply { add(sTipsCookie) }
-            } else {
-                this
+        return parseClientCookiesHeader(cookies)
+            .map { Cookie(it.key, it.value) }
```
Comment (FooIbar) [inline app/src/main/kotlin/com/hippo/ehviewer/client/EhCookieStore.kt:77]: Actually, here as well.
It's used in `copyNecessaryCookies`.

**Pull request merged**